tests/cephfs/cephfs_metrics/cephfs_metrics_scale.py

```diff
@@ -0,0 +1,429 @@
+import json
+import random
+import string
+import time
+import traceback
+
+from ceph.ceph import CommandFailed
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+"""
+Testing CephFS metrics Scale
+1. Client Node1: try to increase "opened_files" and check only designed MDS increases the metric
+2. Client Node2: try to increase "opened_inodes"and "opened_icaps" and "total_inodes"
+3. Client Node2,4: Increase "dentry" realated metrcis
+4. Client Node2 try to increase "total_read_ops" and "total_read_size"
+5. Client Node4 try to increase "total_write_ops" and "total_write_size"
+6. Do "scp" operation from Node1 to Node2
+"""
+
+
+def run(ceph_cluster, **kw):
+    try:
+        tc = "CEPH-83588355"
+        log.info(f"Running CephFS tests for - {tc}")
+
+        # Initialize the utility class for CephFS
+        fs_util = FsUtils(ceph_cluster)
+
+        # Get the client nodes
+        clients = ceph_cluster.get_ceph_objects("client")
+        config = kw.get("config")
+
+        # Authenticate the clients
+        fs_util.auth_list(clients)
+        build = config.get("build", config.get("rhbuild"))
+
+        # Prepare the clients
+        fs_util.prepare_clients(clients, build)
+        client1 = clients[0]
+        client2 = clients[1]
+        client3 = clients[2]
+        client4 = clients[3]
+        # max mds set 2
+        client1.exec_command(sudo=True, cmd="ceph fs set cephfs max_mds 2")
+        # install JQ package for all the clients
+        for client in [client1, client2, client3, client4]:
+            jq_check = client.exec_command(cmd="rpm -qa | grep jq")
+            if "jq" not in jq_check:
+                client.exec_command(sudo=True, cmd="yum install -y jq")
+        cephfs = "cephfs"
+        # Generate random string for directory names
+        rand = "".join(
+            random.choice(string.ascii_lowercase + string.digits) for _ in range(5)
+        )
+        # remove all the fs
+        current_fs, _ = client1.exec_command(sudo=True, cmd="ceph fs ls --format=json")
+        fs_names = json.loads(current_fs)
+        fs_list = [item["name"] for item in fs_names]
+        for fs in fs_list:
+            fs_util.remove_fs(client1, fs)
+        # Create CephFS
+        fs_util.create_fs(client1, cephfs)
+
+        # Define mount directories
+        fuse_mounting_dir_1 = f"/mnt/cephfs_fuse_{rand}_1"
+        fuse_mounting_dir_2 = f"/mnt/cephfs_fuse_{rand}_2"
+        fuse_mounting_dir_3 = f"/mnt/cephfs_fuse_{rand}_3"
+        fuse_mounting_dir_4 = f"/mnt/cephfs_fuse_{rand}_4"
+        cephfs = "cephfs"
+        # Mount CephFS using ceph-fuse and kernel
```

Comment:
I see only fuse_mount and kernel mount are missing. Any reason for having only fuse?

Reply:
@hkadam134 for kernel mount , so far we are not able to find which client is mounted on which directory. 
for fuse-mount we are able to find which client ID is mounted which directory using `session ls` in `mount_point` in `client_metadata`,but I try to find directory using client ID, it does not really provide the `mount_point`. I have mentioned this on `ceph-dev-wq-cicd` channel.

Reply:
Did you cover removal of files/data from the clients and validate if any reduction in counters and validate them ?
-> only reduction happening in the metric is `opened_files` since it shows all the opened files and if you close them all it decreases to zero.

Reply:
Can you also include the check for validating the decreasing value ?

Reply:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-IKSWCP/cephfs_metric_0.log

I have updated the PR that checking "opened_files" metrics go down to 0 when all the processes are killed.